compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -701,6 +701,12 @@ public void visit(BLangListConstructorExpr listConstructor) {
             List<BType> expTypes = tupleType.tupleTypes;
             List<BType> results = new ArrayList<>();
             BType restType = null;
+
+            // [int, boolean, string...] x = [1, true];
+            if (tupleType.restType != null && listConstructor.exprs.size() < expTypes.size() + 1) {
+                restType = tupleType.restType;
```

Comment:
It looks like line 719 overwrite this https://github.com/ballerina-platform/ballerina-lang/pull/19348/files#diff-1f0acc81bd72ee2f8368b0a25cd59e9cR719

Reply:
Yes. However, if L706 satisfies, it won't reach L719. Cause, L706 checks `listConstructor.exprs.size() < expTypes.size() + 1` and if it satisfies, `i < expTypes.size()` will always be true, and won't reach L719.